delivery_multi_destination/views/delivery_carrier_view.xml

```diff
@@ -41,23 +41,23 @@
             <field name="delivery_type" position="after">
                 <field name="destination_type"/>
                 <field name="country_ids"
-                       invisible="not context.get('show_children_carriers', False)"
+                       invisible="not context.get('show_children_fields', False)"
                 />
                 <field name="state_ids"
-                       invisible="not context.get('show_children_carriers', False)"
+                       invisible="not context.get('show_children_fields', False)"
                 />
                 <field name="zip_from"
-                       invisible="not context.get('show_children_carriers', False)"
+                       invisible="not context.get('show_children_fields', False)"
                 />
                 <field name="zip_to"
-                       invisible="not context.get('show_children_carriers', False)"
+                       invisible="not context.get('show_children_fields', False)"
                 />
             </field>
         </field>
     </record>
 
     <record id="delivery.action_delivery_carrier_form" model="ir.actions.act_window">
-        <field name="domain">[('parent_id', '=', False)]</field>
+        <field name="context">{'show_children_carriers': True}</field>
```

Comment:
When the menu option is hit this context is passed, and the default view is tree, so the children are visible in the tree. So I tried to set it to `False` here and to `True` in the form field that loads the tree.

Reply:
Solution is to keep both, domain and context. I have done it.